packages/react-native-bpk-theming/src/BpkThemeAttributes.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow */
+
+import { type Node } from 'react';
+import { colorBlue500 } from 'bpk-tokens/tokens/base.react.native';
+
+import { withTheme } from './BpkThemeProvider';
+
+type Attributes = {
+  primaryColor: string,
+};
+
+export type Props = {
+  children: Attributes => Node,
+  theme: { primaryColor: string },
+};
+
+/** A render prop component that provides certain values
+ * from the nearest `BpkThemeProvider` with suitable defaults
+ * if rendered without a `BpkThemeProvider`.
+ *
+ * Supported attributes:
+ * + `primaryColor`
+ *
+ * @example
+ * <BpkThemeAttributes>
+ *  {({ primaryColor }) => <View style={{ backgroundColor: primaryColor }} />}
+ * </BpkThemeAttributes>
+ *
+ * @param {Object} props `children` prop is used as render prop.
+ * @return {Node} The return value of `props.children`
+ *
+ */
```

Comment:
Whats the deal with JSDoc comment? We don't do this anywhere else...are we now makign a concerted effort to do this? Doesn't flow negate this somewhat?

Reply:
We've been doing them in Objective-C and Java/Kotlin which also has a type system. One doesn't preclude the other I think. Most editors will show this extra information when autocompleting which is nice imo

Reply:
Cool!

I'm a little concerned about them rotting but they are no different to our current `README.md`'s in that regard. I guess its yet another place we have to update too...

Change is hard 😅

Reply:
This might be helpful https://www.npmjs.com/package/eslint-plugin-jsdoc

Reply:
We already have something in our eslint config that enforces writing JSdoc comments correctly however it only triggers when you start writing one